### PR TITLE
Add PostComment API

### LINKED ISSUE #28: Add Comments API
Part of #26.

This can be split out into multiple issues to reflect the endpoints in the API docs: http://docs.codecorpsapidevelop.apiary.io/#reference/categories
- [ ] Create
- [ ] List all
- [ ] Filter
- [ ] Update

### PR DESCRIPTION
**Pull request by npendery**
Closes #28

### CODE REVIEW

File: priv/repo/migrations/20160817220118_create_comment.exs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+defmodule CodeCorps.Repo.Migrations.CreateComment do
+  use Ecto.Migration
+
+  def change do
+    create table(:comments) do
+      add :body, :text
+      add :markdown, :text
+      add :state, :string
```
Comment (joshsmith) [inline priv/repo/migrations/20160817220118_create_comment.exs]: I would make all of these `null: false`.

File: web/models/comment.ex [hunk 2]
```diff
@@ -0,0 +1,33 @@
+defmodule CodeCorps.Comment do
+  use EctoStateMachine,
```
Comment (joshsmith) [inline web/models/comment.ex]: Looks like you left this out of `mix.exs`.

File: web/controllers/comment_controller.ex [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  def delete(conn, %{"id" => id}) do
+    comment = Repo.get!(Comment, id)
+
+    # Here we use delete! (with a bang) because we expect
+    # it to always work (and if it does not, it will raise).
+    Repo.delete!(comment)
+
+    send_resp(conn, :no_content, "")
+  end
```
Comment (joshsmith) [inline web/controllers/comment_controller.ex]: Should remove the `delete` endpoint.

File: test/models/comment_test.exs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+defmodule CodeCorps.CommentTest do
+  use CodeCorps.ModelCase
+
+  alias CodeCorps.Comment
+
+  @valid_attrs %{body: "some content", markdown: "some content", state: "some content"}
```
Comment (joshsmith) [inline test/models/comment_test.exs]: This looks like it it should not be valid since the `state` is not in the set of states we expect.

**Comment by joshsmith** [comment]
Yes, I think even after seeing I'm still partial to calling this `Comment` and not `PostComment`.

**Comment by joshsmith** [comment]
I'll finish this up.

**Comment by rjsamson** [comment]
Looks good to me. 👍

**Pull request merged**